### DISCUSSION TITLE
What readBogusMetaFile is diffrent from readsMetaFile

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15078] [Q&A]
https://github.com/prometheus/prometheus/blob/199d2fb492322af138ef85d08ba022066fea784c/tsdb/repair.go#L126
hoe does repaire work?

**Comment by bboreham** [comment]
As it says in the code: 
```
// repairBadIndexVersion repairs an issue in index and meta.json persistence introduced in
// commit 129773b41a565fde5156301e37f9a87158030443.
```

So this code exists to correct an error which was in [some code from 6 years ago](https://github.com/prometheus/prometheus/commit/129773b41a565fde5156301e37f9a87158030443).
It is not very important today.
